CalibCalorimetry/EBPhase2TPGTools/plugins/EcalEBPhase2TPParamProducer.cc

```diff
@@ -0,0 +1,347 @@
+  }
+  toCompressStream << "\n";
+  tmpStringConv = toCompressStream.str();
+  tmpStringOut = tmpStringConv.c_str();
+  gzwrite(out_file_, tmpStringOut, std::strlen(tmpStringOut));
+  toCompressStream.str(std::string());
+
+  for (unsigned int iGr = 0; iGr < nWeightGroups_; iGr++) {
+    toCompressStream << "WEIGHTTIME " << dec << iGr << std::endl;
+    for (long unsigned int i = 0; i < timeWeights[iGr].size(); i++) {
+      if (timeWeights[iGr][i] < 0)
+        toCompressStream << "-0x" << std::hex << abs(timeWeights[iGr][i]) << " ";
+      else
+        toCompressStream << "0x" << std::hex << timeWeights[iGr][i] << " ";
+    }
+    toCompressStream << "\n";
+  }
+
+  toCompressStream << "\n";
+  tmpStringConv = toCompressStream.str();
+  tmpStringOut = tmpStringConv.c_str();
+  gzwrite(out_file_, tmpStringOut, std::strlen(tmpStringOut));
+  toCompressStream.str(std::string());
+
+  //  fill  map between xTals and groups. If each xTal is a group there is a one-to-one map
+  const std::vector<DetId>& ebCells = theBarrelGeometry->getValidDetIds(DetId::Ecal, EcalBarrel);
+  std::map<int, int> mapXtalToGroup;
+
+  int iGroup = 0;
+  for (const auto& it : ebCells) {
+    EBDetId id(it);
+    std::pair<int, int> xTalToGroup(id.rawId(), iGroup);
+    mapXtalToGroup.insert(xTalToGroup);
+    iGroup++;
+  }
+
+  //write to file
+
+  for (std::map<int, int>::const_iterator it = mapXtalToGroup.begin(); it != mapXtalToGroup.end(); it++) {
+    toCompressStream << "CRYSTAL " << dec << it->first << std::endl;
+    toCompressStream << it->second << std::endl;
+  }
+  tmpStringConv = toCompressStream.str();
+  tmpStringOut = tmpStringConv.c_str();
+  gzwrite(out_file_, tmpStringOut, std::strlen(tmpStringOut));
+  toCompressStream.str(std::string());
+
+  /////////////////////////////////////
+
+  for (const auto& it : ebCells) {
+    EBDetId id(it);
+    toCompressStream << "LINCONST " << dec << id.rawId() << std::endl;
+    double theta = theBarrelGeometry->getGeometry(id)->getPosition().theta();
+    EcalLiteDTUPedestalsMap::const_iterator itped = theEcalTPPedestals->getMap().find(id);
+
+    if (itped != theEcalTPPedestals->end()) {
+      peds = &(*itped);
+
+    } else {
+      edm::LogWarning("EcalEBPhase2TPParamProducer") << " could not find EcalLiteDTUPedestal entry for " << id;
+    }
+
+    int shift, mult;
+    double calibCoeff = 1.;
+    bool ok;
+    int tmpPedByGain;
+    for (unsigned int i = 0; i < ecalPh2::NGAINS; ++i) {
+      ok = computeLinearizerParam(theta, gainRatio_[i], calibCoeff, shift, mult);
+      if (!ok) {
+        edm::LogWarning("EcalEBPhase2TPParamProducer")
+            << "unable to compute the parameters for SM=" << id.ism() << " xt=" << id.ic() << " " << id.rawId();
+
+        shift = 0;
+        tmpPedByGain = 0;
+        mult = 0;
+        toCompressStream << " 0x0"
+                         << " 0x0"
+                         << " 0x0" << std::endl;
+      } else {
+        tmpPedByGain = (int)(peds->mean(i) + 0.5);
```

Comment:
This piece of sofwtare is meant to run on one event to build the parameters. Although it is an unlikely situation, *(.i.e. that peds will not be able to get values from wherever) it would crash in the event the peds pointer is null. Which is exacly what we want. I replaced the LogWarning  with a LogError

Reply:
If the "pedestal not found" is worth of aborting the job (as opposed to skipping that element of `ebCells`), the code should really thrown an exception on line 155 instead of relying on a crash (which is not a guaranteed behavior by the standard).